### PR TITLE
Add OptionalImmutableListEncoding

### PR DESCRIPTION
**Pull request by ianagbip1oti**
Adds convenience builder methods for `Optional<ImmutableList<T>>`

### CODE REVIEW

File: src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.github.princesslana.eriscasper.data.immutable.encoding;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Optional;
+import org.immutables.encode.Encoding;
+import org.immutables.encode.Encoding.StandardNaming;
+
+@Encoding
+public class OptionalImmutableListEncoding<T> {
```
Comment (CoreyShupe) [inline src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java]: Why is this needed but NullableImmutableListEncoding isn't?

Comment (CoreyShupe) [reply src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java]: Otherwise, it looks good.

File: src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  @Encoding.Impl private Optional<ImmutableList<T>> value = Optional.empty();
+
+  @Encoding.Builder
+  static class Builder<T> {
```
Comment (CoreyShupe) [inline src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java]: We should heavily consider having tests for these to make sure the encoding is working. Compile -> Test so we don't need to worry about the compilation procedure. I think it would be a good idea to test the encoding just like we do the Nullable class and Optional differences.

File: src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java (hunk 1 above)
Comment (ianagbip1oti) [reply src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java]: NullableImmutableListEncoding may be needed, but I haven't found a spot where it is yet. So, I haven't bothered with it for this PR.

Comment (CoreyShupe) [reply src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java]: Gotcha
